src/internal/collection/types.go

```diff
@@ -81,37 +82,53 @@ type ReadWriteCollection interface {
 	Upsert(key string, val proto.Message, f func() error) error
 	Create(key string, val proto.Message) error
 	Delete(key string) error
-	DeleteAll()
-	DeleteAllPrefix(prefix string)
+	DeleteAll() error
 }
 
-// ReadWriteIntCollection is a ReadonlyCollection interface specifically for ints.
-type ReadWriteIntCollection interface {
-	Create(key string, val int) error
-	Get(key string) (int, error)
-	Increment(key string) error
-	IncrementBy(key string, n int) error
-	Decrement(key string) error
-	DecrementBy(key string, n int) error
-	Delete(key string) error
+type PostgresReadWriteCollection interface {
```

Comment:
Won't these only ever be implemented by one type? They probably don't need to be interfaces.

Reply:
They don't need to be, but this file serves as a good entry point to see the differences between the interfaces.  There should probably be better doc strings for each call (like you mentioned in the `README.md`), so I can add those in.